build.sbt

```diff
@@ -1,9 +1,10 @@
+import Scalaz._
 
 lazy val root = Project(
   id = "root",
   base = file(".")
 ).settings(
-  scalaVersion := "2.12.2"
+  scalaVersion := "2.12.3"
```

Comment:
If what you want to achieve is to remove duplication then in this particular case I think even better than using AutoPlugin's projectSettings is using [sbt-travisci](https://github.com/dwijnand/sbt-travisci).

More generally, there are lots of ways to write builds - often down to personal preference/habit or niche scaling issues. So I held back to just version upgrades and deprecated Build, and even those are scope-creeping in this PR ;-)